src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.File.Tests.cs

```diff
@@ -206,5 +208,84 @@ public void Add_SymbolicLink(TarEntryFormat format, bool createTarget)
                 Assert.Null(reader.GetNextEntry());
             }
         }
+
+        [Theory]
+        [MemberData(nameof(WriteEntry_CopyArchive_Data))]
+        public void WriteEntry_CopyArchive(TestTarFormat testFormat, string testCaseName, CompressionMethod compressionMethod, bool copyData)
+        {
+            TarEntryFormat entryFormat = GetEntryFormatForTestTarFormat(testFormat);
+            string pathWithExpectedFiles = GetTestCaseUnarchivedFolderPath(testCaseName);
+            if (!Path.EndsInDirectorySeparator(pathWithExpectedFiles))
+            {
+                pathWithExpectedFiles = pathWithExpectedFiles + Path.DirectorySeparatorChar;
+            }
+
+            using Stream file = GetTarMemoryStream(compressionMethod, testFormat, testCaseName);
+            using Stream originArchive = compressionMethod == CompressionMethod.GZip ? new GZipStream(file, CompressionMode.Decompress) : file;
+
+            VerifyCopyArchive(pathWithExpectedFiles, originArchive, entryFormat, copyData);
+        }
+
+        protected void VerifyCopyArchive(string pathWithExpectedFiles, Stream originArchive, TarEntryFormat entryFormat, bool copyData)
+        {
+            TarEntry entry;
+
+            using MemoryStream destinationArchive = new MemoryStream();
+            using (TarReader reader = new TarReader(originArchive, leaveOpen: false))
+            {
+                using (TarWriter writer = new TarWriter(destinationArchive, entryFormat, leaveOpen: true))
+                {
+                    while ((entry = reader.GetNextEntry(copyData)) != null)
+                    {
+                        if (entry is PaxTarEntry paxEntry)
+                        {
+                            paxEntry.GroupName = TestLongGName;
+                            paxEntry.UserName = TestLongUName;
+                        }
+                        writer.WriteEntry(entry);
+                    }
+                }
+            }
+
+            destinationArchive.Position = 0;
+
+            Dictionary<string, FileSystemInfo> expectedEntries = GetFileSystemInfosRecursive(pathWithExpectedFiles).ToDictionary(fsi =>
+                fsi.FullName.Substring(pathWithExpectedFiles.Length).Replace(Path.DirectorySeparatorChar, '/'));
```

Comment:
Another way you can validate the copy is by re-iterating originArchive (along with destinationArchive) and comparing as much fields as possible against the entries in destinationReader.

Reply:
Why not both?

Reply:
I don't see value in extracting the entries and then using them for comparison in this case, the reported error is when you copy a Tar to another without writing to the disk. IMO, the filesystem should not be involved at all.

Reply:
But if there is a bug in the way we are reading information with the Tar APIs, then comparing the items read with the first TarReader with a second TarReader will give me the same results. The utf8 case discussed above is a good example of this.

Reply:
Then there should be a separate test that validates file system entries extracted to disk against tar archive entries.